wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -56,6 +56,12 @@ public final ByteString unknownFields() {
    */
   public abstract Builder newBuilder();
 
+  /** Returns this message with any unknown fields removed. */
+  public final T withoutUnknownFields() {
+    //noinspection unchecked
+    return (T) newBuilder().clearUnknownFields().build();
```

Comment:
The cast and warning here are a result of the raw type above. When they are correctly parameterized both go away.